### PR TITLE
Example showing basic CRUD operations

### PR DESCRIPTION
**Pull request by jrvarma**
This example demonstrates all the `CRUD` operations for a single calendar (specified by its `uid`). It does not contain any user interface, but simply defines a class that exposes methods for each of the `CRUD` operations and for `sync` with the server. The class is initialized with user credentials  and either the encryption password or the cipher key. The calling program (a `CLI`) may obtain these parameters from terminal input or some secure storage (like a key ring) and then create an instance of `EtesyncCRUD` as follows.
```
# call with cipher key
crud = EtesyncCRUD(email, None, remoteUrl, uid, authToken, cipher_key)
# call with encryption password
crud = EtesyncCRUD(email, userPassword, remoteUrl, uid, authToken, None)
```
The calling (`CLI`) program can then invoke `crud.create_event`, `crud.retrieve_event`, `crud.update_event`, `crud.delete_event` and `crud.sync` as needed.

Been testing this for a few days now, and seemed to be working fine. But, yesterday after adding/updating some events and syncing from the first machine, I am getting an integrity exception (HMAC mismatch) on the second machine and on an android device. First machine is syncing fine. 

Am I doing something wrong? How do I debug this? How can I revert the server to an older state or rebuild the server data from the cache on one of the machines? Any suggestions would be appreciated. (I might add a `panic` method in `EtesyncCRUD` for something like this.)

PS: I am storing the `cipher_key` as a base64 encoded string and converting it back to bytes for the `init` call.

### CODE REVIEW

File: example_crud.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+
+import etesync as api
+
+
+class EtesyncCRUD:
+    def __init__(self, email, userPassword, remoteUrl, uid, authToken,
+                 cipher_key=None):
+        r"""Initialize EtesyncInterface
```
Comment (tasn) [inline example_crud.py]: The "r" here isn't needed, it denotes regular expression, which is not what this string is.
(same for the same thing in the rest of the file)

File: example_crud.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        Returns
+        -------
+        True if event created
+        False otherwise (Error is stored in self.error)
+        """
+        try:
+            ev = api.Event.create(self.journal.collection, event)
+            ev.save()
+            return True
+        except Exception as inst:
```
Comment (tasn) [inline example_crud.py]: Why do this and not just let the exception rise? I think it's cleaner to just have the exception there. That's what exceptions are for, exceptions...

File: example_crud.py (hunk 1 above)
Comment (jrvarma) [reply example_crud.py]: OK

File: example_crud.py (hunk 2 above)
Comment (jrvarma) [reply example_crud.py]: OK. This was a left over of the days when this ran as a daemon and daemons cannot raise.

**Comment by jrvarma** [comment]
Any suggestions on the integrity exception? I am unable to sync across devices till this is resolved. Reverting or rebuilding is fine.

**Comment by tasn** [comment]
Sorry, I didn't see your initial message, just the code, my bad!

Integrity: please contact support and I'll help you fix. It looks like there may be a bug with the python api that's causing corruption in some cases and that's why it happened. The code as shown here is fine!

**Comment by jrvarma** [comment]
I wonder whether support is available since I am running etesync on my own server. Anyway, I have emailed the email address (reports@xxx.xxx) suggested by the android app.

**Comment by tasn** [comment]
Self-hosting is going to make it harder to help you, but I'm happy to help when I can. Will reach out there soon.

**Comment by jrvarma** [comment]
Have made changes as suggested and updated the pull request.

File: example_crud.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        Parameters
+        ----------
+        uid : uid of event to be retrieved
+
+        Returns
+        -------
+        On success returns iCalendar file (as a string)
+        Otherwise returns None (Error is stored in self.error)
+        """
+        try:
```
Comment (tasn) [inline example_crud.py]: You forgot an exception here

File: example_crud.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            return None
+
+    def all_events(self):
+        """Retrieve all events in calendar
+
+        Returns
+        -------
+        List of iCalendar files (as strings)
+        """
+        return [E.content for E in self.calendar.list()]
```
Comment (tasn) [inline example_crud.py]: sorry for not noticing this before, but can you use lower case e rather than a capital to conform with python conventions?

**Pull request merged**